### ISSUE TITLE
💣 otel span metrics stop working

### ISSUE THREAD

**Issue by jojotong** [kubegems/kubegems#368]
prometheus `calls_total` metrics disapreared, and log in otel collector:
```log
$ kubectl logs -f -n observability opentelemetry-collector-66f4dd8d68-ms5qz --tail 100
2022-12-23T02:06:53.106Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-alarm-service\\x00/alarm/subscribe/page\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00roc-alarm-service-85f7d49548-rvrs7\\x00200\""}
2022-12-23T02:06:53.307Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-multitenant-service\\x00/actuator/prometheus\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00stable-roc-multitenant-service-6b5f4d7464-cwxgm\\x00200\""}
2022-12-23T02:06:53.509Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-alarm-service\\x00/alarm/subscribe/page\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00roc-alarm-service-85f7d49548-rvrs7\\x00200\""}
2022-12-23T02:06:53.711Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-alarm-service\\x00/alarm/subscribe/page\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00roc-alarm-service-85f7d49548-rvrs7\\x00200\""}
2022-12-23T02:06:53.913Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"smartvoice-portal\\x00prepipeline-dmkit\\x00SPAN_KIND_INTERNAL\\x00STATUS_CODE_UNSET\\x00smartvoice-cloudpepper-v3-dit\\x00stable-smartvoice-portal-748d76f4d6-h278d\""}
2022-12-23T02:06:54.115Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-alarm-service\\x00/alarm/subscribe/page\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00roc-alarm-service-85f7d49548-rvrs7\\x00200\""}
2022-12-23T02:06:54.316Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-alarm-service\\x00/alarm/subscribe/page\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00roc-alarm-service-85f7d49548-rvrs7\\x00200\""}
2022-12-23T02:06:54.919Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-alarm-service\\x00/alarm/subscribe/page\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00roc-alarm-service-85f7d49548-rvrs7\\x00200\""}
2022-12-23T02:06:55.322Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"smartvoice-portal\\x00prepipeline-dmkit\\x00SPAN_KIND_INTERNAL\\x00STATUS_CODE_UNSET\\x00smartvoice-cloudpepper-v3-dit\\x00stable-smartvoice-portal-748d76f4d6-h278d\""}
2022-12-23T02:06:55.524Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-multitenant-service\\x00/actuator/prometheus\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00stable-roc-multitenant-service-6b5f4d7464-cwxgm\\x00200\""}
2022-12-23T02:06:57.130Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-multitenant-service\\x00/actuator/prometheus\\x00SPAN_KIND_SERVER\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00stable-roc-multitenant-service-6b5f4d7464-cwxgm\\x00200\""}
2022-12-23T02:06:57.532Z        warn    batchprocessor/batch_processor.go:178   Sender failed   {"kind": "processor", "name": "batch", "pipeline": "traces", "error": "value not found in metricKeyToDimensions cache by key \"roc-multitenant-service\\x00OperatorController.getOperatorDetailByIdoeToken\\x00SPAN_KIND_INTERNAL\\x00STATUS_CODE_UNSET\\x00harix-roc-cloudpepper-v3-dit\\x00stable-roc-multitenant-service-6b5f4d7464-cwxgm\""}
```

**Comment by jojotong** [comment]
According to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9018. 
We should upgrade otel collector to v0.59+, now it's v0.58.

**Comment by jojotong** [comment]
fixd at https://github.com/kubegems/plugins/commit/0721e55e1555d9a53d8e9b001992f528611991bb

**Comment by LinkMaq** [comment]
v0.59+ still has the same problem and it consistent with this issue. https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9018#issuecomment-1252476417

>I've just removed my workarounds to reproduce it. There's no logs of an error. But I do see cpu spiking when this happen.

![image](https://user-images.githubusercontent.com/2688646/209762481-3738ecb6-7196-45dc-903e-8142d08e93ef.png)

**Comment by jojotong** [comment]
The CPU spiking and the failure of the otel collector are two problems. This issue only discusses how to solve the failure of the otel collector.

**Comment by jojotong** [comment]
We can only debug when problems such as the loss of the otel data occur.

**Comment by jojotong** [comment]
Recurrence by set `dimensions_cache_size: 1`, it always appears after the metrics key cache is full, Expecially when cpu usage high.

So I think after we use `prometheusexporter` to replace `prometheusremotewriteexporter`, it will be fine.

If not, we can wait for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16601 https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17307 to be merge, then use the new release.

**Comment by jojotong** [comment]
PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17307 has been released in v0.70.0 https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.70.0.

So we upgrade opentemetry plugin: https://github.com/kubegems/plugins/commit/75b19474de7caf128b99dd79f704c951f1befd0b.

**Comment by jojotong** [comment]
Seems no problem after upgrade, close.
